yt_dlp/extractor/niconico.py

```diff
@@ -367,248 +385,169 @@ def parse_format_id(id_code):
-            api_data,
-            (lambda x: x['channel']['name'],
-             lambda x: x['community']['name']))
-
-        # Note: cannot use api_data.get('owner', {}) because owner may be set to "null"
-        # in the JSON, which will cause None to be returned instead of {}.
-        owner = try_get(api_data, lambda x: x.get('owner'), dict) or {}
-        uploader_id = str_or_none(
-            get_video_info_web(['ch_id', 'user_id'])
-            or owner.get('id')
-            or channel_id
-        )
-        uploader = (
-            get_video_info_web(['ch_name', 'user_nickname'])
-            or owner.get('nickname')
-            or channel
-        )
+        tags = None
+        if webpage:
+            # use og:video:tag (not logged in)
+            og_video_tags = re.finditer(r'<meta\s+property="og:video:tag"\s*content="(.*?)">', webpage)
+            tags = list(filter(None, (clean_html(x.group(1)) for x in og_video_tags)))
+            if not tags:
+                # use keywords and split with comma (not logged in)
+                kwds = self._html_search_meta('keywords', webpage, default=None)
+                if kwds:
+                    tags = [x for x in kwds.split(',') if x]
+        if not tags:
+            # find in json (logged in)
+            tags = traverse_obj(api_data, ('tag', 'items', ..., 'name'))
+
+        subtitles = None
+        if self.get_param('writesubtitles', False):
+            comment_user_key = traverse_obj(api_data, ('comment', 'keys', 'userKey'))
+            user_id_str = session_api_data.get('serviceUserId')
+
+            thread_ids = [x for x in traverse_obj(api_data, ('comment', 'threads')) if x['isActive']]
+            raw_danmaku = self._extract_all_comments(video_id, thread_ids, user_id_str, comment_user_key)
+            if raw_danmaku:
+                raw_danmaku = json.dumps(raw_danmaku)
+
+                subtitles = {
+                    'comments': [{
+                        'ext': 'json',
+                        'data': raw_danmaku,
+                    }],
+                }
+            else:
+                self.report_warning(f'Failed to get comments. {bug_reports_message()}')
 
         return {
             'id': video_id,
             '_api_data': api_data,
-            'title': title,
+            'title': get_video_info(('originalTitle', 'title')) or self._og_search_title(webpage, default=None),
             'formats': formats,
-            'thumbnail': thumbnail,
-            'description': description,
-            'uploader': uploader,
-            'timestamp': timestamp,
-            'uploader_id': uploader_id,
-            'channel': channel,
-            'channel_id': channel_id,
-            'view_count': view_count,
-            'comment_count': comment_count,
-            'duration': duration,
-            'webpage_url': webpage_url,
+            'thumbnail': traverse_obj(api_data, ('video', 'thumbnail', 'url')) or self._html_search_meta(
```

Comment:
```suggestion
            'thumbnail': get_video_info('thumbnail', 'url') or self._html_search_meta(
```
this was the whole point of the change to get_video_info!